### PR TITLE
Added convenience functions to add elastic network restraints by chain

### PR DESCRIPTION
**Pull request by splenz**
I've added a few functions to use the indexing in MELD to add elastic network restraints within a protein chain. Let me know what changes you would like me to make before we merge. I've tested this on a few small systems. I've also compared the output restraints from this function to our old functions outside MELD, and they are identical for the systems I've tested it on. A few things that I think we may need to discuss are:

- Indexing. Currently, the functions don't use the chain id relative residue indexing. This is because the residue indices are extracted from the openmm topology object that uses absolute indexing. There might be a better way of indexing that I've overlooked
- Organization. I created a new file for this and placed it in builders, but maybe it's better to place the functions in an existing file.
- Adding restraints. Right now, there is a separate function that adds the restraints to an input system object. Maybe we'd prefer it to return the collection object and have the user add the collection in the setup file.

### CODE REVIEW

File: meld/system/builders/build_elastic_network_restraints.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+from meld.system import restraints
+from meld.system import meld_system
+from meld.system import scalers
+from meld import unit as u
+
+def create_elastic_network_restraints(
+    system: meld_system.System,
+    scaler: scalers.RestraintScaler,
+    chain_index: int,
+    cutoff: float = 1.0,
```
Comment (jlmaccal) [inline meld/system/builders/build_elastic_network_restraints.py:24]: Can we make this a Quantity, and add checks to make sure that units are included? If you look at the code for some of the restraints, there should be examples of how to do this.

File: meld/system/builders/build_elastic_network_restraints.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+from meld.system import restraints
+from meld.system import meld_system
+from meld.system import scalers
+from meld import unit as u
+
+def create_elastic_network_restraints(
+    system: meld_system.System,
+    scaler: scalers.RestraintScaler,
+    chain_index: int,
```
Comment (jlmaccal) [inline meld/system/builders/build_elastic_network_restraints.py:23]: I was thinking that chain_index could also be an iterable. So you could either specify a single chain, or multiple chains. In the case of multiple chains, you would be creating the joint set of restraints between both chains, e.g. to keep a multimer together. If you wanted to separately create restraints for each chain, you would just call this function multiple times, once with each chain.

**Review by jlmaccal — CHANGES_REQUESTED** [review]
Looks good. A few minor things to tweak and maybe an improvement to make (multiple chains).

File: meld/system/builders/build_elastic_network_restraints.py (hunk 2 above)
Comment (splenz) [reply meld/system/builders/build_elastic_network_restraints.py]: I've come up with something that works for specifying multiple chain indices. I'm not sure if it's the best/most pythonic way of doing things, so please let me know if there is a better way.

**Review by jlmaccal — APPROVED** [review]

**Pull request merged**